Intro docs reorg (#510)
CitrineInformatics/citrine-python: docs/source/getting_started/overview.rst

========
Overview
========

The Citrine python client serves as the entry point for creating and interacting with resources on the Citrine Platform.
There are a few basic types of resources to become acquainted with: :doc:`projects <projects>`, :doc:`datasets <datasets>`, :doc:`data objects <data_model>`, and :doc:`AI modules <ai_modules>`.
A project is a container for a thematically related group of datasets and AI modules that a specific group of people have access to.

> A dataset contains a related set of data objects: information about materials samples, the way they were processed, and the measurements performed on them.
> A dataset can also contain files related to these data objects.

Reviewer: ```suggestion
A dataset is an access control unit for data objects and files: information about materials samples, the way they were processed, and the measurements performed on them.
```
